components/base/sensorcontrolled/movestraight.go

```diff
@@ -43,13 +43,26 @@ func (sb *sensorBase) MoveStraight(
 		}
 		return sb.controlledBase.MoveStraight(ctx, distanceMm, mmPerSec, extra)
 	}
+	if sb.position == nil {
+		sb.logger.CWarn(ctx,
+			"Position reporting sensor not available, controlling using linear velocity only")
+
+		// adjust inputs to ensure errDist is always positive to match the position based implementation
+		if distanceMm < 0 {
+			distanceMm = -distanceMm
+			mmPerSec = -mmPerSec
+		}
+	}
 
 	// make sure the control loop is enabled
 	if sb.loop == nil {
 		if err := sb.startControlLoop(); err != nil {
 			return err
 		}
 	}
+
+	// pause and resume the loop to reset the control blocks
+	sb.loop.Pause()
```

Comment:
the current sensor controlled spin/movestraight will "kick" when a request is made in the opposite direction of the previous, so I added this to reset the control blocks

Reply:
thanks for adding the comment